### PR TITLE
[FEATURE] Add latest post plugin with configurable limit

### PR DESCRIPTION
**Pull request by haassie**
By adding the settings via a flexform, no additional code has to be
changed. The settings in the flexform is now working identical to the
option via TypoScript.

Resolves: #85

### CODE REVIEW

**Comment by benjaminkott** [comment]
@haassie can we make a new element for this? 
that would make it more easy to provide custom templates for that

**Comment by haassie** [comment]
Ah, that was the reason we always did it with a separate element... Will do!

File: Resources/Private/Templates/Post/ListLatestPosts.html [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<f:layout name="Default"/>
+<f:section name="Content">
+    <f:render partial="List" arguments="{_all}"/>
+</f:section>
+<f:section name="FooterAssets">
```
Comment (benjaminkott) [inline Resources/Private/Templates/Post/ListLatestPosts.html]: i dont think that we need a comment form on the latest post template?

File: ext_localconf.php [hunk 2]
```diff
@@ -25,6 +25,13 @@
             'Post' => 'listRecentPosts',
         ]
     );
+    \TYPO3\CMS\Extbase\Utility\ExtensionUtility::configurePlugin(
+        'T3G.AgencyPack.Blog',
+        'LatestPosts',
```
Comment (benjaminkott) [inline ext_localconf.php]: is it latest or recent?

File: Classes/Controller/PostController.php [hunk 3]
```diff
@@ -156,6 +156,20 @@ public function listRecentPostsAction(): void
...
+    /**
+     * Show a number of latest posts.
+     *
+     * @throws \TYPO3\CMS\Extbase\Configuration\Exception\InvalidConfigurationTypeException
+     * @throws \TYPO3\CMS\Extbase\Persistence\Exception\InvalidQueryException
+     */
+    public function listLatestPostsAction(): void
+    {
+        $maximumItems = (int) ($this->settings['lists']['posts']['numberOfLatestItems'] ?: 3);
+        $posts = $this->postRepository->findAllWithLimit($maximumItems);
```
Comment (benjaminkott) [inline Classes/Controller/PostController.php]: i think we should exclude the current post from this list

File: Configuration/FlexForms/LatestPosts.xml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    <meta>
+        <langDisable>1</langDisable>
+    </meta>
+    <sheets>
+
+        <sDEF>
+            <ROOT>
+                <TCEforms>
+                    <sheetTitle>
+                        LLL:EXT:blog/Resources/Private/Language/locallang_db.xlf:display_flexform.sheet_general
```
Comment (benjaminkott) [inline Configuration/FlexForms/LatestPosts.xml]: remove linebreaks and move into single line

File: Configuration/FlexForms/LatestPosts.xml [hunk 5]
```diff
@@ -0,0 +1,32 @@
+<T3DataStructure>
+    <meta>
+        <langDisable>1</langDisable>
+    </meta>
+    <sheets>
+
```
Comment (benjaminkott) [inline Configuration/FlexForms/LatestPosts.xml]: remove empty line

File: Configuration/FlexForms/LatestPosts.xml [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+                            <config>
+                                <type>input</type>
+                                <eval>int</eval>
+                            </config>
+                        </TCEforms>
+                    </settings.lists.posts.numberOfLatestItems>
+                </el>
+            </ROOT>
+        </sDEF>
+
```
Comment (benjaminkott) [inline Configuration/FlexForms/LatestPosts.xml]: remove empty line

File: Configuration/FlexForms/LatestPosts.xml [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+                    <sheetTitle>
+                        LLL:EXT:blog/Resources/Private/Language/locallang_db.xlf:display_flexform.sheet_general
+                    </sheetTitle>
+                </TCEforms>
+                <type>array</type>
+                <el>
+                    <settings.lists.posts.numberOfLatestItems>
+                        <TCEforms>
+                            <label>
+                                LLL:EXT:blog/Resources/Private/Language/locallang_db.xlf:display_flexform.maximumDisplayedItems
```
Comment (benjaminkott) [inline Configuration/FlexForms/LatestPosts.xml]: remove linebreaks and move into single line

File: Configuration/TypoScript/Static/setup.typoscript [hunk 8]
```diff
@@ -49,6 +49,7 @@ plugin.tx_blog {
             }
             posts {
                 maximumDisplayedItems = {$plugin.tx_blog.settings.lists.posts.maximumDisplayedItems}
+                numberOfLatestItems = {$plugin.tx_blog.settings.lists.posts.numberOfLatestItems}
```
Comment (benjaminkott) [inline Configuration/TypoScript/Static/setup.typoscript]: would more likely move this configuration into a dedicated upper section like `recent-/latestposts`

**Pull request merged**